src/chat/__tests__/fetchingReducer-test.js

```diff
@@ -14,134 +14,91 @@ import { MESSAGE_FETCH_START, MESSAGE_FETCH_ERROR } from '../../actionConstants'
+        eg.action.register_complete,
+      ].reduce(fetchingReducer, eg.baseReduxState.fetching),
+    ).toEqual(eg.baseReduxState.fetching);
+  });
+
   describe('MESSAGE_FETCH_START', () => {
     test('if messages are fetched before or after the corresponding flag is set', () => {
-      const initialState = deepFreeze({
-        [HOME_NARROW_STR]: { older: false, newer: false },
-      });
-
-      const action = deepFreeze({
-        type: MESSAGE_FETCH_START,
-        narrow: HOME_NARROW,
-        numBefore: 10,
-        numAfter: 10,
-      });
-
-      const expectedState = {
-        [HOME_NARROW_STR]: { older: true, newer: true },
-      };
-
-      const newState = fetchingReducer(initialState, action);
-
-      expect(newState).toEqual(expectedState);
+      const prevState = deepFreeze({ [HOME_NARROW_STR]: { older: false, newer: false } });
+      expect(
+        fetchingReducer(
+          prevState,
+          deepFreeze({
+            type: MESSAGE_FETCH_START,
+            narrow: HOME_NARROW,
+            numBefore: 10,
+            numAfter: 10,
+          }),
+        ),
+      ).toEqual({ [HOME_NARROW_STR]: { older: true, newer: true } });
     });
 
     test('if key for narrow does not exist, it is created and corresponding flags are set', () => {
-      const initialState = deepFreeze({
-        [HOME_NARROW_STR]: { older: false, newer: false },
-      });
-
       const narrow = streamNarrow(eg.stream.stream_id);
-      const action = deepFreeze({
-        type: MESSAGE_FETCH_START,
-        narrow,
-        numBefore: 10,
-        numAfter: 0,
-      });
 
-      const expectedState = {
+      const prevState = deepFreeze({ [HOME_NARROW_STR]: { older: false, newer: false } });
+      expect(
+        fetchingReducer(
+          prevState,
+          deepFreeze({ type: MESSAGE_FETCH_START, narrow, numBefore: 10, numAfter: 0 }),
+        ),
+      ).toEqual({
         [HOME_NARROW_STR]: { older: false, newer: false },
         [keyFromNarrow(narrow)]: { older: true, newer: false },
-      };
-
-      const newState = fetchingReducer(initialState, action);
-
-      expect(newState).toEqual(expectedState);
+      });
     });
 
     test('if fetching for a search narrow, ignore', () => {
-      const initialState = deepFreeze({
-        [HOME_NARROW_STR]: {
-          older: false,
-          newer: false,
-        },
-      });
-
-      const action = deepFreeze({
-        ...eg.action.message_fetch_start,
-        narrow: SEARCH_NARROW('some query'),
-      });
-
-      const newState = fetchingReducer(initialState, action);
-
-      expect(newState).toEqual(initialState);
+      const prevState = deepFreeze({ [HOME_NARROW_STR]: { older: false, newer: false } });
+      expect(
+        fetchingReducer(
+          deepFreeze({ [HOME_NARROW_STR]: { older: false, newer: false } }),
+          deepFreeze({ ...eg.action.message_fetch_start, narrow: SEARCH_NARROW('some query') }),
+        ),
+      ).toEqual(prevState);
```

Comment:
This argument to `fetchingReducer` is meant to be `prevState`, right? (In order to express that we're ignoring this action.)